hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/HdfsNode.java

```diff
@@ -93,20 +88,20 @@ private ExecutorInfo createExecutor(String taskIdName, String nodeName, String e
       .setName(nodeName + " executor")
       .setExecutorId(ExecutorInfoBuilder.createExecutorId("executor." + taskIdName))
       .addAllResources(getExecutorResources())
-      .setCommand(CommandInfoBuilder.createCmdInfo(cmd, getCmdUriList(), getExecutorEnvironment()))
+      .setCommand(CommandInfoBuilder.createCmdInfo(cmd, getCmdUriList(nodeId), getExecutorEnvironment()))
       .build();
   }
 
-  private List<CommandInfo.URI> getCmdUriList() {
+  private List<CommandInfo.URI> getCmdUriList(String nodeId) {
```

Comment:
I don't think hosting the node type as part of our REST API server is the best way, and especially the parsing logic above is quite complex. Maybe there is something I'm missing. We can certainly discuss.